core/src/main/java/org/everit/json/schema/internal/JSONPrinter.java

```diff
@@ -84,4 +86,13 @@ public <K> void printSchemaMap(Map<K, Schema> input) {
         });
         endObject();
     }
+    public void describePropertyDependencies(Map<String, Set<String>> propertyDependencies) throws JSONException {
```

Comment:
I would keep it in its original place. The JSONPrinter is just responsible for printing json. It shouldn't be concerned about how some more complex schema structures are printed. In other words, this class just helps printing json, and shouldn't have any behavior that is specific to json schema.